runtime/sema/type_tags.go

```diff
@@ -935,8 +935,11 @@ func commonSuperTypeOfComposites(types []Type) Type {
 	}
 
 	if hasCommonInterface {
+		if len(commonInterfacesList) == 0 {
+			panic(errors.NewUnreachableError())
+		}
+
 		return &IntersectionType{
-			Type:  superType,
```

Comment:
I think the initialization of `superType` can now be moved below, to the end of this function